server.js

```diff
@@ -139,8 +140,31 @@ const auspicePaths = [
   "/community/:user/:repo/*",
 ];
 
+/*
+ * Redirect to translations of narratives if the client has
+ * set language preference and the translation is available
+ *
+ * Currently only used for nCoV narratives and requires the
+ * language code to be the second to last part of the url.
+ */
+const redirectToTranslatedNarratives = async (req, res) => {
+  const acceptedLanguages = req.acceptsLanguages();
+  if (acceptedLanguages && !acceptedLanguages[0].startsWith("en")) {
+    const {source, prefixParts} = helpers.splitPrefixIntoParts(req.url);
+    prefixParts.splice(-1, 0, acceptedLanguages[0]);
+    const potentialNarrative = prefixParts.join("/");
+    const availableNarratives = await source.availableNarratives();
+    if (availableNarratives.includes(potentialNarrative)) {
+      res.redirect(301, "/narratives/" + potentialNarrative);
```

Comment:
301 is Moved Permanently, which will cause most browsers to remember the redirection and never request the first URL again. It is appropriate for when resources move, but not for something dynamic like language detection. 302 Found is a more appropriate redirection status that won't lead to weird bugs.